### DISCUSSION TITLE
Adding a new IS category for empty JSONs?

### DISCUSSION THREAD

**Discussion by diego898** [TiddlyWiki/TiddlyWiki5#5664] [Ideas]
I was working with `<<jsontiddlers>>` and discovered that it will always return a valid JSON object, even the minimum one: `[]`. I had to explicitly check for this. 

Is it worth adding another category to `is`? Something link `emptyJSON`? (terrible name I know). 

Right now, `blank` checks for empty strings, `emptyJSON` could check for an empty JSON, and `empty` could check for either? 

I made my own `emptyJSON` category (which should be renamed to `empty` if we go with the above description) by just cloning `$:/core/modules/filters/is/blank.js` and replacing the conditions to be: `if((title) && (title != '[]'))` and `if(!(same as prev))` and it seems to work. 

Is this worth pursuing?

**Comment by pmario** [comment]
> I was working with <> and discovered that it will always return a valid JSON object, even the minimum one: `[]`. I had to explicitly check for this.

The 5th element in your post is missing. You need to mark it as `code` so we can see it

**Reply by diego898** [reply]
Thanks Mario - I fixed that

**Comment by pmario** [comment]
What is your usecase? Can you be mores specific here?

Is it generic enough to fit into the core?

**Reply by diego898** [reply]
I was writing a proof-of-concept for an "Archive" for the Projectify plugin. I posted about it on the project page [here](https://github.com/NicolasPetton/Projectify/issues/70), and on our Google Groups [here](https://groups.google.com/g/tiddlywiki/c/nAmu7_Zcxcs/m/DqHBA3wYBAAJ).

In short, I wanted to "flatten" a bunch of todos into a single tiddler after I am done with them. To do so, I appended them as JSONs (using the `<<jsontiddlers>>` macro) to an Archive tiddler, but I didn't want to append a bunch of empty `[]`. I figured this would be captured by `is[blank]` but after reading docs, it isn't. 

As for whether it belongs in the core, the only argument I can give is "for completion’s sake", since we have a blank for strings.

**Reply by pmario** [reply]
>As for whether it belongs in the core, the only argument I can give is "for completion’s sake", since we have a blank for strings.

I could live with that ;)
